fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexQueryPlan.java

```diff
@@ -138,6 +149,58 @@ public static LuceneIndexQueryPlan merge(LuceneIndexQueryPlan plan1, LuceneIndex
         return plan;
     }
 
+    /**
+     * Override here to have specific logic to build the {@link QueryResult} for lucene auto complete suggestion result.
+     */
+    @Nonnull
+    @Override
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull FDBRecordStoreBase<M> store,
+                                                                      @Nonnull EvaluationContext context,
+                                                                      @Nullable byte[] continuation,
+                                                                      @Nonnull ExecuteProperties executeProperties) {
+        if (scanType != IndexScanType.BY_LUCENE_AUTO_COMPLETE) {
+            return super.executePlan(store, context, continuation, executeProperties);
+        } else {
+            final RecordMetaData metaData = store.getRecordMetaData();
+            final Index index = metaData.getIndex(indexName);
+            final Collection<RecordType> recordTypes = metaData.recordTypesForIndex(index);
+            if (recordTypes.size() != 1) {
```

Comment:
if no lucene index should do this then I think it should be a check for all lucene indexes.